### PR TITLE
Move ReflectionIT out of LAPI IT

### LINKED ISSUE #2856: Port ReflectionIT to the Ledger API Test Tool
Port `ReflectionIT` to the Ledger API Test Tool.

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
Closes #2856 

Please don't merge until #3573 is merged and this is rebased on `master`, but feel free to approve. :wink: 

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags, if appropriate
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/sandbox/src/test/suite/scala/com/digitalasset/platform/sandbox/services/reflection/ReflectionIT.scala [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+  "Reflection service" when {
+
+    "accessed" should {
+
+      "provide a list of exposed services" in {
+        for {
+          response <- execRequest(listServices)
+        } yield {
+          response.getListServicesResponse.getServiceCount shouldEqual 13
```
Comment (stefanobaghino-da) [inline ledger/sandbox/src/test/suite/scala/com/digitalasset/platform/sandbox/services/reflection/ReflectionIT.scala]: @SamirTalwar I'm aware this is in conflict with #3573, I'll wait until that's merged to rebase and change this.

Comment (SamirTalwar) [reply ledger/sandbox/src/test/suite/scala/com/digitalasset/platform/sandbox/services/reflection/ReflectionIT.scala]: 🖖

File: ledger/sandbox/src/test/suite/scala/com/digitalasset/platform/sandbox/services/reflection/ReflectionIT.scala [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+          override def onCompleted(): Unit = {
+            doneP.tryFailure(new NoSuchElementException("Stream closed without any response."))
+            ()
+          }
+        })
+    ro.onNext(request)
+    ro.onCompleted()
+    val doneF = doneP.future
+    doneF
+  }
```
Comment (SamirTalwar) [inline ledger/sandbox/src/test/suite/scala/com/digitalasset/platform/sandbox/services/reflection/ReflectionIT.scala]: Why not use `SingleItemObserver` here?

Comment (stefanobaghino-da) [reply ledger/sandbox/src/test/suite/scala/com/digitalasset/platform/sandbox/services/reflection/ReflectionIT.scala]: Makes sense, thanks, addressed by https://github.com/digital-asset/daml/pull/3582/commits/1c27b192828386dfe96162628254003c1f8dde29

File: ledger/sandbox/src/test/suite/scala/com/digitalasset/platform/sandbox/services/reflection/ReflectionIT.scala [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    }
+  }
+
+  private def fileBySymbolReq(symbol: String) =
+    ServerReflectionRequest.newBuilder().setFileContainingSymbol(symbol).build()
+
+  private def execRequest(request: ServerReflectionRequest) = {
+    val observer = new SingleItemObserver[ServerReflectionResponse](_ => true)
+    ServerReflectionGrpc.newStub(channel).serverReflectionInfo(observer).onNext(request)
+    observer.result.map(_.get)
```
Comment (SamirTalwar) [inline ledger/sandbox/src/test/suite/scala/com/digitalasset/platform/sandbox/services/reflection/ReflectionIT.scala]: You can use `SingleItemObserver.first` here.

```scala
    SingleItemObserver[ServerReflectionResponse].first(
      ServerReflectionGrpc
        .newStub(channel)
        .serverReflectionInfo(_)
        .onNext(request)
    )

Comment (stefanobaghino-da) [reply ledger/sandbox/src/test/suite/scala/com/digitalasset/platform/sandbox/services/reflection/ReflectionIT.scala]: Thanks, much better, I struggled with it for a few minutes and gave up. Thanks for keeping the quality up. :+1:

Comment (stefanobaghino-da) [reply ledger/sandbox/src/test/suite/scala/com/digitalasset/platform/sandbox/services/reflection/ReflectionIT.scala]: Addressed by https://github.com/digital-asset/daml/pull/3582/commits/020dc833d392c091979e7e9552d935fd1a2fc125

File: ledger/sandbox/src/test/suite/scala/com/digitalasset/platform/sandbox/services/reflection/ReflectionIT.scala (hunk 1 above)
Comment (stefanobaghino-da) [reply ledger/sandbox/src/test/suite/scala/com/digitalasset/platform/sandbox/services/reflection/ReflectionIT.scala]: Rebased.

**Pull request merged**